api/services/EmailService.js

```diff
@@ -1,50 +1,54 @@
 /**
  * Created by preciousr on 21/01/2016.
  */
-var request = require('request'),
+const request = require('request');
 
-emailService = {
-  submissionConfirmation: function(email,application_reference, send_information, user_ref, serviceType){
-    var url = '/confirm-submission';
-    var postData= {to: email, application_reference: application_reference, send_information: send_information, user_ref: user_ref, service_type: serviceType};
-
-    // send request to notification service
-    request(setOptions(postData, url), function (err, res, body) {
-      if(err) {
-        console.log(err);
-      } else {
-        console.log(res.statusCode, body);
-      }
-    });
-  },
-  failedDocuments: function(email,failed_certs){
-    var url = '/failed-documents';
-    var postData= {to: email, failed_certs: failed_certs};
+const EmailService = {
+    submissionConfirmation(
+        email,
+        application_reference,
+        send_information,
+        user_ref,
+        serviceType
+    ) {
+        const url = '/confirm-submission';
+        const postData = {
+            to: email,
+            application_reference: application_reference,
+            send_information: send_information,
+            user_ref: user_ref,
+            service_type: serviceType,
+        };
+        EmailService._sendRequestToNotificationService(postData, url);
+    },
+    failedDocuments(email, failed_certs) {
+        const url = '/failed-documents';
+        const postData = { to: email, failed_certs: failed_certs };
+        EmailService._sendRequestToNotificationService(postData, url);
+    },
 
-    // send request to notification service
-    request(setOptions(postData, url), function (err, res, body) {
-      if(err) {
-        console.log(err);
-      } else {
-        console.log(res.statusCode, body);
-      }
-    });
-  }
+    _sendRequestToNotificationService(postData, url) {
+        request(setOptions(postData, url), (err, res, body) => {
+            if (err) {
+                sails.log.error(err);
+            } else {
+                sails.log.info(res.statusCode, body);
+            }
+        });
+    },
 };
 
-module.exports = emailService;
-
-function setOptions(postData, url){
-  var options = {
-    url: sails.config.customURLs.notificationServiceURL + url,
-    headers:
-    {
-      'cache-control': 'no-cache',
-      'content-type': 'application/json'
-    },
-    method: 'POST',
-    json: true,
-    body: postData
-  };
-  return options;
+function setOptions(postData, url) {
```

Comment:
it looks like url is unchanged (it's always `/confirm-submission`?), so it might be better to remove this param from the `setOptions` function. 

you might also want to look at this https://www.npmjs.com/package/request#convenience-methods. It looks like you can set a default request, so you don't have to use a function to provide your defaults 🤓 

(also as an aside, request is deprecated, I'm pretty sure node will provide the same functionality `http.request`)

Reply:
oh oops, yes, it looks like you do use setOptions with two different urls. It still might be worth digging into the convenience methods though

Reply:
oops, just some more context
`request(setOptions(postData, url), (err, res, body)` is (imo) really hard to read/understand. You've got to understand what setOptions does with postData/url, and it feels cluttered.. I can't explain it 😅. 

By using a default request, it would look something like this (just some pseudo code..) 
```
const emailRequest = request({
baseUrl: sails.config.customURLs.notificationServiceURL
headers,
method: 'POST',
json: true
})

//inside your method(s) you would make the request like so: 
emailRequest({url: '/confirm-submission', body: ...}) //the rest of it is defaulted
```